Fix unexpected behavior in UG due to invalid HTML markup

tlylt opened the pull request:
**What is the purpose of this pull request?**

- [x] Documentation update
- [x] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] Others, please explain:

<!--
  If this pull request is addressing an issue, link to the issue: "Fixes #xxx" or "Resolves #xxx"

  If the pull request completely addresses the issue, use one of the issue closing keywords. https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords

  Otherwise, elaborate further on the rationale of this pull request as needed
-->

**Overview of changes:**
Fixes #1794
Also fixes #1840 (very much related so putting it together here)

- Include a 'troubleshooting' section in UG for putting such errors

**Anything you'd like to highlight / discuss:**
Not too sure why, but sometimes it's fine for block-level elements to be in `<span>` but sometimes not.
E.g. in [embed](https://markbind.org/userGuide/formattingContents.html#embeds) section, seems like this span tag works fine and does not trigger any vue warning.
![image](https://user-images.githubusercontent.com/41845017/158917005-c607e87f-7e89-49d7-b5b1-5cb504f172d6.png)

**Testing instructions:**
- verify the behavior according to issue #1794
- preview the [added section](https://deploy-preview-1837--markbind-master.netlify.app/userguide/troubleshooting)

<!--
  Any special testing instructions **not including** our automated tests.
-->

**Proposed commit message: (wrap lines at 72 characters)**
Fix unexpected behavior in UG due to invalid HTML markup

Invalid HTML markup may cause Vue hydration issues.

Let's replace span tags with div tags to avoid invalid
inclusion of block-level elements in span tags.

This prevents possible errors in HTML syntax.

<!--
  See this link for more info on how to write a good commit message:
  https://oss-generic.github.io/process/docs/FormatsAndConventions.html#commit-message

  As best as possible, write a succinct commit title in 50 characters
-->

---

**Checklist:** :ballot_box_with_check:

<!-- Leave non-applicable items unchecked -->

- [x] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No unrelated changes <!-- Its tempting, but increases the reviewer's work, and really pollutes the commit history =( -->

<!--
  We'll try our best to get to your PR within a week.
  If we haven't gotten to it then, or if your pull request resolves an urgent item, feel free to give us a ping!
-->

ryoarmanda commented on docs/userGuide/troubleshooting.md:
```diff
@@ -0,0 +1,38 @@
+
+<panel header="Example Underlying Error" type="seamless">
+
+vue.js:634 [Vue warn]: The client-side rendered virtual DOM tree is not matching server-rendered content. This is likely caused by incorrect HTML markup, for example nesting block-level elements inside `<p>`, or missing `<tbody>`. Bailing hydration and performing full client-side render.
```
We can format the example error a little bit nicer, maybe a fenced code block if possible?

ryoarmanda reviewed (commented):
Interesting issue 👀 I'm also a bit intrigued on the embed case, maybe the issue is for specific block-level elements like tables? (cc @jonahtanjz) I think we can also change those to use div for posterity sake, what do you think?

The docs you have written up are a good start, maybe some little stylization nits below:

jonahtanjz commented:
> I'm also a bit intrigued on the embed case, maybe the issue is for specific block-level elements like tables? (cc @jonahtanjz)

Having block elements within inline elements should not be allowed according to the [SSR guide for Vue](https://vuejs.org/guide/scaling-up/ssr.html#hydration-mismatch). For some reason, Vue does not always warn or have issues with some of the cases.

For example, putting the following code on a MarkBind site,
```.html
<span>
<div>Test</div>
</span>
```
doesn't seem to have any warnings.

Similarly, even using the example provided on [Vue's page](https://vuejs.org/guide/scaling-up/ssr.html#hydration-mismatch),
```.html
<p>
<div>Test</div>
</p>
```
no warnings as well.

Putting a table in a `<span>` seems to consistently cause the warning to show up.

The following seems to cause the warning to show up as well,
```.html
<p>
<ul>
<li>1</li>
<li>2</li>
</ul>
</p>
```


> I think we can also change those to use div for posterity sake, what do you think?

Although some of the offending components don't result in any warnings, I think they should still be changed as well since they are not valid HTML and may cause some problematic issues in the future.

jonahtanjz reviewed (commented):
Thanks @tlylt! 

A couple more invalid HTML:

* [ ] [docs/index.md#19](https://github.com/MarkBind/markbind/blob/master/docs/index.md?plain=1#L9) (`<p>` in `<span>`)
* [ ] [docs/userGuide/addingPages.md#17](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/addingPages.md?plain=1#L17) (`<p>` in `<span>`)
* [ ] [docs/userGuide/deployingTheSite.md#17](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/deployingTheSite.md?plain=1#L17) (`<p>` in `<span>`)
* [ ] [docs/userGuide/formattingContents.md#17](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/formattingContents.md?plain=1#L17) (`<p>` in `<span>`)
* [ ] [docs/userGuide/makingTheSiteSearchable.md#19](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/makingTheSiteSearchable.md?plain=1#L19) (`<p>` in `<span>`)
* [ ] [docs/userGuide/markBindInTheProjectWorkflow.md#16](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/markBindInTheProjectWorkflow.md?plain=1#L16) (`<p>` in `<span>`)
* [ ] [docs/userGuide/markBindSyntaxOverview.md#17](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/markBindSyntaxOverview.md?plain=1#L17) (`<p>` in `<span>`)
* [ ] [docs/userGuide/reusingContents.md#17](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/reusingContents.md?plain=1#L17) (`<p>` in `<span>`)
* [ ] [docs/userGuide/siteJsonFile.md#10](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/siteJsonFile.md?plain=1#L10) (`<p>` in `<span>`)
* [ ] [docs/userGuide/themes.md#16](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/themes.md?plain=1#L16) (`<p>` in `<span>`)
* [ ] [docs/userGuide/usingHtmlJavaScriptCss.md#34](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/usingHtmlJavaScriptCss.md?plain=1#L34) (example used is not valid HTML as it will create a `<p>` in `<span>`)
* [ ] [docs/userGuide/syntax/tree.md#96 and #110](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/syntax/tree.md?plain=1#L96) (codeblock and `<tree>` in `<span>`)
* [ ] [docs/userGuide/syntax/popovers.md#48](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/syntax/popovers.md?plain=1#L48) (`<div>` in `<p>`)

Also noticed a few of our Vue components having block-level elements in `<span>`. This can be fixed in another PR.

[Popover.vue](https://github.com/MarkBind/markbind/blob/master/packages/vue-components/src/Popover.vue#L2)
[MinimalPanel.vue](https://github.com/MarkBind/markbind/blob/master/packages/vue-components/src/panels/MinimalPanel.vue#L2)
[NestedPanel.vue](https://github.com/MarkBind/markbind/blob/master/packages/vue-components/src/panels/NestedPanel.vue#L2)

tlylt commented:
> A couple more invalid HTML:

Good catch! I have fixed the cases mentioned. I also:
- added [one warning](https://deploy-preview-1837--markbind-master.netlify.app/userguide/reusingcontents#includes) in the reusing content -> includes section to inform users to choose div over span
- removed some span to div replacement as those spans within a boilerplate appear to be important for the variable to work. Will open a separate issue for this.

jonahtanjz reviewed (approved):
Thanks @tlylt! Looks good to me 👍 

Let's wait for @ryoarmanda to look through and approve as well :)

ryoarmanda reviewed (approved):
Looks good to me as well! Will merge tonight

ryoarmanda commented:
Some weird thing happening here 🤔 the checks have been in queue for a while now, and I didn't see it ever running. Even the checks tab show 0 checks, and now workflow records appear.

If this keeps going, I might have to close and reopen this PR to kick the checks back up again.

tlylt commented:
> Some weird thing happening here 🤔 the checks have been in queue for a while now, and I didn't see it ever running. Even the checks tab show 0 checks, and no workflow records appear.
> 
> 
> 
> If this keeps going, I might have to close and reopen this PR to kick the checks back up again.

GitHub is experiencing some operational error, we can probably wait out.

jonahtanjz commented:
> If this keeps going, I might have to close and reopen this PR to kick the checks back up again.

Will need to try this since it is still stuck in queue.

ryoarmanda commented:
Okay, checks are running now.
